tools/c7n_azure/c7n_azure/resources/sqlserver.py

```diff
@@ -23,3 +33,89 @@ class resource_type(ArmResourceManager.resource_type):
         service = 'azure.mgmt.sql'
         client = 'SqlManagementClient'
         enum_spec = ('servers', 'list', None)
+
+
+@SqlServer.filter_registry.register('firewall-rules')
+class SqlServerFirewallRulesFilter(Filter):
+    """Filters SQL servers by the firewall rules
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: servers-with-firewall
+                resource: azure.sqlserver
+                filters:
+                  - type: firewall-rules
+                    include:
+                    - '131.107.160.2-131.107.160.3'
+                    - 10.20.20.0/24
+    """
+
+    schema = type_schema(
+        'firewall-rules',
+        **{
+            'include': {'type': 'array', 'items': {'type': 'string'}},
```

Comment:
You can do this to avoid extra checks in validate.
```
'oneOf': [
{ 'include': {'type': 'array', 'items': {'type': 'string'}}},
{ 'equal': {'type': 'array', 'items': {'type': 'string'}}}
]
```

Reply:
I think it enforces both checks you have: requires one, blocks both

Reply:
If I use both the load throws:
Error on policy:test-azure-sql-server resource:azure.sqlserver
Additional properties are not allowed ('equal', 'include' were unexpected)
that is hard to understand

Reply:
It doesn't work exactly like that. I looked for samples the best I could do is:
`schema = {
    'type': 'object',
    # 'notify': {'type': 'array'}, # what should it be?
    'properties': {
        'include': {'type': 'array', 'items': {'type': 'string'}},
        'equal': {'type': 'array', 'items': {'type': 'string'}},
    },
    'oneOf': [
        {'required': ['include']},
        {'required': ['equal']}
    ]
}`
Which still passes through the case (doesn't generate an error) when none of the properties were specified. The validation does work if both are present but the error message is worse than what I currently have:
`Error on policy:test-azure-sql-server resource:azure.sqlserver
{'type': 'firewall-rules', 'equal': [], 'include': []} is valid under each of {'required': ['equal']}, {'required': ['include']}

Failed validating 'oneOf' in schema[2]:
    {'oneOf': [{'required': ['include']}, {'required': ['equal']}],
     'properties': {'equal': {'items': {'type': 'string'}, 'type': 'array'},
                    'include': {'items': {'type': 'string'},
                                'type': 'array'}},
     'type': 'object'}

On instance:
    {'equal': [], 'include': [], 'type': 'firewall-rules'}`

Reply:
discussing in #3907 .. there's some other schema thats triggering. in the case none of required  present. finding out which schema that is unfortunately fairly complex.